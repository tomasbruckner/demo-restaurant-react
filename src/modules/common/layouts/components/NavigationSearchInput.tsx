import React, { PureComponent } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from 'core/redux/store';
import {
  getSearchSelector,
  searchResultFilterAsync,
  clearSearch,
} from 'core/redux/ducks/SearchDuck';
import SearchIcon from '@material-ui/icons/Search';
import { fade, withStyles } from '@material-ui/core/styles';
import styled from 'styled-components/macro';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import Autocomplete from '@material-ui/lab/Autocomplete';
import CircularProgress from '@material-ui/core/CircularProgress';
import Input from 'modules/common/components/controls/Input';
import Typography from 'modules/common/components/controls/Typography';
import { RouteComponentProps, withRouter } from 'react-router';
import { getRestaurantDetail } from 'core/router/urls';
import { DEFAULT_SEARCH_PAGE_SIZE } from 'utils/constants';

const SearchIconWrapper = styled.div`
  padding-left: 16px;
  padding-right: 16px;
  height: 100%;
  position: absolute;
  pointer-events: none;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const SearchWrapper = styled.div`
  position: relative;
  border-radius: 3px;
  margin-left: 32px;
  background-color: ${p => {
    return `${fade(p.theme.white, 0.15)}`;
  }};
  '&:hover': {
    background-color: ${p => `${fade(p.theme.white, 0.25)}`};
  },
  margin-left: 0;
  width: 300px;
`;

const StyledInput = withStyles({
  root: {
    color: 'white',
    '&:focus': {
      color: 'white',
    },
  },
})(Input);

const connector = connect(
  (state: RootState) => {
    return {
      searchResult: getSearchSelector(state),
    };
  },
  {
    dispatchSearch: searchResultFilterAsync,
    dispatchClearSearch: clearSearch,
  },
);

class NavigationSearchInput extends PureComponent<
  WithTranslation & ConnectedProps<typeof connector> & RouteComponentProps
> {
  state = {
    isLoading: false,
    isOpen: false,
    searchValue: '',
  };

  public componentDidMount() {
    const { dispatchClearSearch } = this.props;

    dispatchClearSearch();
  }

  onToggleOpen = () => {
    const { isOpen } = this.state;

    this.setState({ isOpen: !isOpen, searchValue: '' });
  };

  redirect = (restaurantId: number) => {
    const { history } = this.props;

    history.push(getRestaurantDetail(restaurantId));
  };

  onChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
    const { dispatchSearch } = this.props;

    this.setState({ isLoading: true, searchValue: event.target.value });
    try {
      await dispatchSearch({
        take: DEFAULT_SEARCH_PAGE_SIZE,
        skip: 0,
        sort: {
          predicate: 'name',
          reverse: false,
        },
        filter: {
          name: event.target.value,
        },
      });
    } finally {
      this.setState({ isLoading: false });
    }
  };

  render() {
    const { searchResult } = this.props;
    const { isLoading, isOpen, searchValue } = this.state;

    return (
      <SearchWrapper>
        {!isOpen && (
          <SearchIconWrapper>
            <SearchIcon />
          </SearchIconWrapper>
        )}
        <Autocomplete
          style={{ width: 300 }}
          open={isOpen}
          onOpen={this.onToggleOpen}
          onClose={this.onToggleOpen}
          options={searchResult.data}
          loading={isLoading}
          getOptionLabel={o => o.name}
          clearOnBlur
          clearOnEscape
          inputValue={searchValue}
          renderOption={option => (
            <Typography onClick={() => this.redirect(option.restaurantId)}>
              {option.name}
            </Typography>
          )}
          renderInput={params => (
            <StyledInput
              {...params}
              fullWidth
              variant="outlined"
              onChange={this.onChange}
              InputProps={{
                ...params.InputProps,
                style: {
                  color: 'white',
                },
                endAdornment: (
                  <>
                    {isLoading ? <CircularProgress color="inherit" size={20} /> : null}
                    {params.InputProps.endAdornment}
                  </>
                ),
              }}
            />
          )}
        />
      </SearchWrapper>
    );
  }
}

export default withTranslation(I18N_CORE)(connector(withRouter(NavigationSearchInput)));
