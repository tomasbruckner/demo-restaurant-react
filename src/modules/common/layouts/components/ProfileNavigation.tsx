import React, { PureComponent } from 'react';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import Link from 'modules/common/utils/Link';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import { LOGIN, USER_PROFILE } from 'core/router/urls';
import { RouteComponentProps, withRouter } from 'react-router';
import { logOut } from 'core/redux/ducks/UserDuck';
import { connect, ConnectedProps } from 'react-redux';

const connector = connect(null, {
  dispatchLogOut: logOut,
});

class ProfileNavigationComponent extends PureComponent<
  WithTranslation & ConnectedProps<typeof connector> & RouteComponentProps
> {
  state = {
    anchorEl: null,
  };

  onMenuClose = () => {
    this.setState({ anchorEl: null });
  };

  onProfileClick = ({ currentTarget }: React.SyntheticEvent) => {
    this.setState({ anchorEl: currentTarget });
  };

  onLogoutClick = () => {
    const { history, dispatchLogOut } = this.props;

    dispatchLogOut();
    history.push(LOGIN);
  };

  render() {
    const { anchorEl } = this.state;
    const { t } = this.props;

    return (
      <>
        <IconButton
          edge="end"
          aria-label="account of current user"
          aria-haspopup="true"
          color="inherit"
          onClick={this.onProfileClick}
        >
          <AccountCircle />
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
          keepMounted
          transformOrigin={{ vertical: 'top', horizontal: 'right' }}
          open={!!anchorEl}
          onClose={this.onMenuClose}
        >
          <MenuItem to={USER_PROFILE} component={Link}>
            {t('component.navigation.menu.profile')}
          </MenuItem>
          <MenuItem onClick={this.onLogoutClick}>{t('component.navigation.menu.logout')}</MenuItem>
        </Menu>
      </>
    );
  }
}

export default withTranslation(I18N_CORE)(connector(withRouter(ProfileNavigationComponent)));
