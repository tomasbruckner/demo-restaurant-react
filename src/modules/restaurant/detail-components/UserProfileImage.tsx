import React, { PureComponent } from 'react';
import AccountCircle from '@material-ui/icons/AccountCircle';
import Avatar from '@material-ui/core/Avatar';
import { withStyles } from '@material-ui/core';
import styled from 'styled-components/macro';
import { API_ENDPOINT } from 'config/config';

const StyledAvatar = withStyles({
  root: {
    height: '100%',
    width: '100%',
    marginBottom: '32px',
  },
})(Avatar);

const StyledAccountCircle = withStyles({
  root: {
    height: '100%',
    width: '100%',
  },
})(AccountCircle);

const StyledWrapper = styled.div`
  max-height: 100px;
  max-width: 100px;
  margin-right: 16px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

type Props = {
  imageLink?: string;
};

export class UserProfileImage extends PureComponent<Props> {
  render() {
    const { imageLink } = this.props;

    return (
      <StyledWrapper>
        <StyledAvatar src={imageLink ? `${API_ENDPOINT}${imageLink}` : undefined}>
          <StyledAccountCircle />
        </StyledAvatar>
      </StyledWrapper>
    );
  }
}

export default UserProfileImage;
