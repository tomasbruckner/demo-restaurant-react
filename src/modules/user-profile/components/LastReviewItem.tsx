import React, { PureComponent } from 'react';
import { ReviewResponse } from 'services/reviewService';
import ReviewDetail from 'modules/restaurant/detail-components/ReviewDetail';
import styled from 'styled-components/macro';

type Props = {
  review: ReviewResponse;
};

const StyledWrapper = styled.div`
  margin: 16px;
`;

export class LastReviewItem extends PureComponent<Props> {
  render() {
    const { review } = this.props;

    return (
      <StyledWrapper>
        <ReviewDetail data={review} withRestaurantLink />
      </StyledWrapper>
    );
  }
}

export default LastReviewItem;
