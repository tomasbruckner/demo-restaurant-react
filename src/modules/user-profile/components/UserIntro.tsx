import React, { ComponentType, PureComponent } from 'react';
import { UserResponse } from 'services/userService';
import { Card, CardContent, withStyles } from '@material-ui/core';
import Typography from 'modules/common/components/controls/Typography';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';
import { formatDate } from 'utils/commonUtils';
import PersonAddIcon from '@material-ui/icons/PersonAdd';
import MailOutlineIcon from '@material-ui/icons/MailOutline';
import FaceIcon from '@material-ui/icons/Face';
import BeenHereIcon from '@material-ui/icons/Beenhere';
import styled from 'styled-components/macro';

const StyledTypography = withStyles({
  root: {
    marginBottom: '16px',
  },
})(Typography);

const getStyledIcon = (Component: ComponentType) =>
  withStyles({
    root: {
      marginRight: '8px',
    },
  })(Component);

const StyledPersonAddIcon = getStyledIcon(PersonAddIcon);
const StyledMailOutlineIcon = getStyledIcon(MailOutlineIcon);
const StyledFaceIcon = getStyledIcon(FaceIcon);
const StyledBeenHereIcon = getStyledIcon(BeenHereIcon);

const StyledFlexDiv = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 8px;
`;

type Props = {
  user: UserResponse;
};

export class UserIntro extends PureComponent<Props & WithTranslation> {
  render() {
    const {
      t,
      user: { created, roleId, email, firstName, lastName },
    } = this.props;

    return (
      <Card variant="outlined">
        <CardContent>
          <StyledTypography variant="h5">{t('component.user-intro.title')}</StyledTypography>
          <StyledFlexDiv>
            <StyledFaceIcon />
            <Typography display="inline" variant="body2" component="p">
              {`${firstName} ${lastName}`}
            </Typography>
          </StyledFlexDiv>
          <StyledFlexDiv>
            <StyledMailOutlineIcon />
            <Typography display="inline" variant="body2" component="p">
              {email}
            </Typography>
          </StyledFlexDiv>
          <StyledFlexDiv>
            <StyledBeenHereIcon />
            <Typography display="inline" variant="body2" component="p">
              {t(`component.user-intro.role-${roleId}`)}
            </Typography>
          </StyledFlexDiv>
          <StyledFlexDiv>
            <StyledPersonAddIcon />
            <Typography display="inline" variant="body2" component="p">
              {t('component.user-intro.joined', { date: formatDate(created) })}
            </Typography>
          </StyledFlexDiv>
        </CardContent>
      </Card>
    );
  }
}

export default withTranslation(I18N_CORE)(UserIntro);
