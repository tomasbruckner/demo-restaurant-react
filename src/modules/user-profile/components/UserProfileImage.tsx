import React, { PureComponent } from 'react';
import AccountCircle from '@material-ui/icons/AccountCircle';
import Avatar from '@material-ui/core/Avatar';
import { withStyles } from '@material-ui/core';
import styled from 'styled-components/macro';
import FileUploadDialog from 'modules/common/components/file-upload/FileUploadDialog';
import ImageService from 'services/imageService';
import { connect, ConnectedProps } from 'react-redux';
import { userInfoAsync } from 'core/redux/ducks/UserDuck';
import { API_ENDPOINT } from 'config/config';

const StyledAvatar = withStyles({
  root: {
    height: '100%',
    width: '100%',
    marginBottom: '32px',
    cursor: 'pointer',
  },
})(Avatar);

const StyledAccountCircle = withStyles({
  root: {
    height: '100%',
    width: '100%',
  },
})(AccountCircle);

const StyledWrapper = styled.div`
  max-height: 150px;
  max-width: 150px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: auto;
`;

const connector = connect(null, {
  dispatchGetMe: userInfoAsync,
});

type Props = {
  imageLink?: string;
};

export class UserProfileImage extends PureComponent<Props & ConnectedProps<typeof connector>> {
  state = {
    isUploadDialogOpen: false,
  };

  onToggleUploadDialog = () => {
    const { isUploadDialogOpen } = this.state;

    this.setState({ isUploadDialogOpen: !isUploadDialogOpen });
  };

  onUploadFile = async (file: File) => {
    const { dispatchGetMe } = this.props;
    const formData = new FormData();
    formData.append('file', file);

    this.onToggleUploadDialog();
    await ImageService.uploadUserProfileImage(formData);
    dispatchGetMe();
  };

  render() {
    const { isUploadDialogOpen } = this.state;
    const { imageLink } = this.props;

    return (
      <StyledWrapper>
        <StyledAvatar
          src={`${API_ENDPOINT}${imageLink}`}
          onClick={this.onToggleUploadDialog}
          style={{ height: '100%' }}
        >
          <StyledAccountCircle />
        </StyledAvatar>
        <FileUploadDialog
          onSubmit={this.onUploadFile}
          isOpen={isUploadDialogOpen}
          onClose={this.onToggleUploadDialog}
        />
      </StyledWrapper>
    );
  }
}

export default connector(UserProfileImage);
