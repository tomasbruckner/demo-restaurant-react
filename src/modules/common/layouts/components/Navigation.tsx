import React, { PureComponent } from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from 'modules/common/components/controls/Typography';
import Link from 'modules/common/utils/Link';
import styled from 'styled-components/macro';
import { WithTranslation, withTranslation } from 'react-i18next';
import { I18N_CORE } from 'core/i18n/i18n';
import { OWNED_RESTAURANTS, RESTAURANTS, USER_MANAGEMENT } from 'core/router/urls';
import ProfileNavigation from './ProfileNavigation';
import NavigationSearchInput from './NavigationSearchInput';
import RoleAccessRenderer from '../../utils/RoleAccessRenderer';

const FlexGrow = styled.div`
  flex-grow: 1;
`;

const LinkWrapper = styled.div<Props>`
  display: flex;
  margin-left: 32px;
`;

const StyledLink = styled(Link)`
  color: white;
  margin-right: 16px;
  text-decoration: none;
  &:visited {
    color: white;
  }

  &:active {
    color: white;
  }

  &:hover {
    color: white;
  }

  &:link {
    color: white;
  }
`;

type Props = {};

class NavigationComponent extends PureComponent<Props & WithTranslation> {
  render() {
    const { t } = this.props;

    return (
      <FlexGrow>
        <AppBar position="static">
          <Toolbar>
            <Typography variant="h6" noWrap>
              {t('component.navigation.home')}
            </Typography>
            <NavigationSearchInput />
            <div />
            <LinkWrapper>
              <Typography component={StyledLink} to={RESTAURANTS}>
                {t('component.navigation.links.restaurants')}
              </Typography>
              <RoleAccessRenderer adminOrOwner>
                <Typography component={StyledLink} to={OWNED_RESTAURANTS}>
                  {t('component.navigation.links.owned-restaurants')}
                </Typography>
              </RoleAccessRenderer>
              <RoleAccessRenderer onlyAdmin>
                <Typography component={StyledLink} to={USER_MANAGEMENT}>
                  {t('component.navigation.links.users')}
                </Typography>
              </RoleAccessRenderer>
            </LinkWrapper>
            <FlexGrow />
            <div>
              <ProfileNavigation />
            </div>
          </Toolbar>
        </AppBar>
      </FlexGrow>
    );
  }
}

export default withTranslation(I18N_CORE)(NavigationComponent);
