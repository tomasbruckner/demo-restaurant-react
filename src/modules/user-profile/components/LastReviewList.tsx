import React, { PureComponent } from 'react';
import { PagedDto } from 'types/commonTypes';
import { ReviewResponse } from 'services/reviewService';
import LastReviewItem from './LastReviewItem';

type Props = {
  reviews: PagedDto<ReviewResponse>;
};

export class LastReviewList extends PureComponent<Props> {
  render() {
    const { reviews } = this.props;

    return reviews.data.map(review => <LastReviewItem review={review} />);
  }
}

export default LastReviewList;
