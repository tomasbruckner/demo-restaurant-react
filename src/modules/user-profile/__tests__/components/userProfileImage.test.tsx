import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { UserProfileImage } from 'modules/user-profile/components/UserProfileImage';
import { fakeFunction, fakeTranslationProps } from 'utils/testUtils';

describe('UserProfileImage', () => {
  it('basic example', () => {
    const tree = renderer
      .create(
        <UserProfileImage
          dispatchGetMe={fakeFunction}
          imageLink="link"
          {...fakeTranslationProps}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
