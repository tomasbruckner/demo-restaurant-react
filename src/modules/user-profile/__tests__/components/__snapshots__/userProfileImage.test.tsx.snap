// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileImage basic example 1`] = `
.c0 {
  max-height: 150px;
  max-width: 150px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
}

<div
  className="c0"
>
  <div
    className="MuiAvatar-root WithStyles(ForwardRef(Avatar))-root-1 MuiAvatar-circle"
    onClick={[Function]}
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <img
      className="MuiAvatar-img"
      src="https://jest:9999/link"
    />
  </div>
  <div />
</div>
`;
